### ISSUE TITLE
More false positives for C416

### ISSUE THREAD

**Issue by mwgamble** [adamchainz/flake8-comprehensions#493]
### Python Version

3.10, 3.11, 3.12

### flake8 Version

5.0.4

### Package Version

3.11.1

### Description

I'm seeing a false positive for C416 after upgrading from 3.10.1 to 3.11.1. The code looks like this:

```python
import string
from enum import Enum
from typing import Final

class Thing(Enum):
    ABC = "abc"
    DEF = "def"
    GHI = "ghi"
    JKL = "jkl"

X_TO_Y_MAPPING: Final = {
    key: value
    for key, value in zip(
        chunks(tuple(string.ascii_lowercase), len(string.ascii_lowercase) // 4), Thing
    )
}
```

I tried rewriting the code like this:

```python
X_TO_Y_MAPPING: Final = dict(
    (key, value)
    for key, value in zip(
        chunks(tuple(string.ascii_lowercase), len(string.ascii_lowercase) // 4), Thing
    )
)
```

But then I got the following error:

```
C402 Unnecessary generator - rewrite as a dict comprehension.
```

And that just takes me back to where I started :(

**Comment by adamchainz** [comment]
Rewrite as:

```python
dict(zip(
    chunks(tuple(string.ascii_lowercase), len(string.ascii_lowercase) // 4), Thing
))
```

I think the doc'd example is clear enough, but if you have any suggestions let me know.

**Comment by mwgamble** [comment]
Thank you :heart: In hindsight that should have been obvious :pensive:
